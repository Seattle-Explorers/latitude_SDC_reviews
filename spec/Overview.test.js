import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { mount } from 'enzyme';

import 'babel-polyfill';

import mockData from '../__mocks__/mockData.js'; //eslint-disable-line
import Overview from '../client/src/components/Overview.jsx'; //eslint-disable-line

describe('tests for overview component', () => {
  const { reviews, ...rest } = mockData;

  test('should render', () => {
    const component = <Overview overview={rest} />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should display average and reviews count', () => {
    const component = mount(<Overview overview={rest} />);
    const outputText = component.find('div').text();

    expect(outputText).toBe('3.08 (6 reviews)');
  });
});
