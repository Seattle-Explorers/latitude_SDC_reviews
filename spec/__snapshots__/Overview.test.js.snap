// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for overview component should render 1`] = `
.c0 {
  font-size: 1.5em;
  font-weight: bolder;
  margin-left: 50px;
  margin-bottom: 15px;
}

<div
  className="c0"
>
  2.05
   (
  7
   reviews)
</div>
`;
